src/sentry/workflow_engine/handlers/condition/slow_condition_query_handlers.py

```diff
@@ -180,45 +169,76 @@ def get_rate_bulk(
         return result
 
 
-class BaseEventFrequencyCountHandler:
-    comparison_json_schema = {
-        "type": "object",
-        "properties": {
-            "interval": {"type": "string", "enum": list(STANDARD_INTERVALS.keys())},
-            "value": {"type": "integer", "minimum": 0},
-        },
-        "required": ["interval", "value"],
-        "additionalProperties": False,
-    }
-
-    @staticmethod
-    def evaluate_value(value: WorkflowJob, comparison: Any) -> DataConditionResult:
-        if len(value.get("snuba_results", [])) != 1:
-            return False
-        return value["snuba_results"][0] > comparison["value"]
-
-
-class BaseEventFrequencyPercentHandler:
-    comparison_json_schema = {
-        "type": "object",
-        "properties": {
-            "interval": {"type": "string", "enum": list(STANDARD_INTERVALS.keys())},
-            "value": {"type": "integer", "minimum": 0},
-            "comparison_interval": {"type": "string", "enum": list(COMPARISON_INTERVALS.keys())},
-        },
-        "required": ["interval", "value", "comparison_interval"],
-        "additionalProperties": False,
-    }
-
-    @property
-    def intervals(self) -> dict[str, tuple[str, timedelta]]:
-        return PERCENT_INTERVALS
-
-    @staticmethod
-    def evaluate_value(value: WorkflowJob, comparison: Any) -> DataConditionResult:
-        if len(value.get("snuba_results", [])) != 2:
-            return False
-        return (
-            percent_increase(value["snuba_results"][0], value["snuba_results"][1])
-            > comparison["value"]
+slow_condition_query_handler_registry = Registry[type[BaseEventFrequencyQueryHandler]](
+    enable_reverse_lookup=False
+)
+
+
+@slow_condition_query_handler_registry.register(Condition.EVENT_FREQUENCY_COUNT)
+@slow_condition_query_handler_registry.register(Condition.EVENT_FREQUENCY_PERCENT)
+class EventFrequencyQueryHandler(BaseEventFrequencyQueryHandler):
```

Comment:
I really like the composition here -- letting us add more conditions here and then handling the results in different ways feels like a great place for the abstraction.